config/initializers/rack_cors.rb

```diff
@@ -8,29 +8,45 @@
 # Read more: https://github.com/cyu/rack-cors
 
 Rails.application.config.middleware.insert_before 0, Rack::Cors do
+  max_age = 1.day.to_i
+  expose  = %w[
+    keygen-accept-signature
+    keygen-signature
+    keygen-date
+    keygen-digest
+    keygen-account-id
+    keygen-account
+    keygen-bearer-id
+    keygen-bearer
+    keygen-token-id
+    keygen-token
+    x-ratelimit-window
+    x-ratelimit-count
+    x-ratelimit-limit
+    x-ratelimit-remaining
+    x-ratelimit-reset
+    x-request-id
+    x-signature
+    date
+    digest
+  ]
+
+  options = {
+    headers: :any,
+    methods: :any,
+    max_age:,
+    expose:,
+  }
+
+  allow do
+    origins { _1.ends_with?(Keygen::DOMAIN) }
+
+    resource '*', **options, credentials: true
```

Comment:
We could add [a `X-CSRF-Token` header](https://security.stackexchange.com/questions/154410/how-does-x-csrf-token-work) and use that as an additional means of [preventing CSRF](https://developer.mozilla.org/en-US/docs/Web/Security/Practical_implementation_guides/CSRF_prevention#implement_an_anti-csrf_token_along_with_samesitestrict).